broadinstitute/single_cell_portal_core
# Enable cell filtering in IGV via URL, fix a search case (SCP-5673)

This lets users link to filtered IGV states, which helps share epigenomic insights.  It also fixes a bug with search and IGV.

### Overview
[Previously](https://github.com/broadinstitute/single_cell_portal_core/pull/2075), users could instantly filter tracks, but they could not persist or share those filtered results.  Also, a bug caused IGV to error when the user loaded IGV, then changed to a no-gene or multi-gene view, then returned to any single-gene IGV view.

Now, cell filtering results can be saved in IGV by simply copying the current URL.  Loading the URL restores that filtered IGV state.  This makes it easy to share chromatin accessibility insights with collaborators and research audiences.

### Video
Here's how it looks!

https://github.com/user-attachments/assets/bf01965f-0abd-4caa-a506-697f5055639e

### Test
To manually test, if you'd like:
1.  Go to a study that has an ATAC fragment BED file, e.g. the 3k cell PBMC 10x multiome example dataset
* Staging: https://singlecell-staging.broadinstitute.org/single_cell/study/SCP395
* Production: https://singlecell.broadinstitute.org/single_cell/study/SCP2699
2.  Search gene SPIB
3. Go to "Genome" tab
4. Click "Filter plotted cells"
5. Click root checkbox in "Atac cluster" facet, to deselect all filters
6. Click filter for "5" 
7. Confirm ATAC fragment track updates
8. Click the "<-" button to the right of the gene search box
9. In the default, 0-gene view, search for ANXA1
10. Confirm "Genome" tab is in view, and IGV shows your new gene and the ATAC track
11. Clear your filters
12. Confirm ATAC track updates
13. Apply a filter
14. Confirm ATAC track updates
15. Refresh page
16. Confirm ATAC track shows state from step 14 (unfiltered ATAC might very briefly appear; unideal but expected)

Also note that these changes required updating to the latest upstream IGV, which e.g. changes the genes track to show Ensembl transcript ID instead of gene name / symbol.  The familiar gene name is still visible in the search bar and related genes ideogram.  As a separate task, it might help to refine IGV or our UI so users can easily discern which genes are in the genomic frame, which becomes more pertinent if the user drags or zooms.

This satisfies SCP-5673.
